### PR TITLE
Implement jax2tf scatter_* ops with no XLA

### PR DESCRIPTION
**Pull request by oliverdutton**
Linking to #9269

This is a work in progress, it seems to work for the simplest examples. Will update with tests, etc.

### CODE REVIEW

**Comment by oliverdutton** [comment]
I'm doing this to convert some JAX code to ONNX. I find that my current fix using
```python
tensor_scatter_nd_add
tensor_scatter_nd_max
tensor_scatter_nd_min
```

don't get converted by tf2onnx, while `tensor_scatter_nd_update` works fine and my devolved implementation of `scatter_mul` in JAX converts correctly. So I'm going to reimplement `add,max,min` similarly to how `mul` is currently

**Comment by oliverdutton** [comment]
Reimplementation looks prettier in the onnx representation

Pic below is scatter_add.
image

**Review by gnecula — CHANGES_REQUESTED** [review]
Please see the suggestions about testing in #9269

File: jax/experimental/jax2tf/impl_no_xla.py [hunk 1]
```diff
@@ -791,10 +791,70 @@ def _dynamic_update_slice(operand, update, *start_indices,
...
+    mode,
+    _in_avals: Sequence[core.ShapedArray],
+    _out_aval: core.ShapedArray):
+    # Does the updates in operand shapes, inefficient
+    scatter_indices_flat = tf.squeeze(scatter_indices, -1)
+    updates = segment_op(updates, scatter_indices_flat, operand.shape[0])
+    return update_op(operand, updates)
+  return _dense_scatter
+
+for (xla_op, segment_op, update_op) in [
```
[the remark this answers is not in the document]
Comment (oliverdutton) [reply jax/experimental/jax2tf/impl_no_xla.py]: Probably best unrolled.

**Comment by oliverdutton** [comment]
I have found other routes for the task than no_xla, closing task. Thank you for your suggestions and comments.

**Comment by marcvanzee** [comment]
@oliverdutton for reference, what other route did you find?

**Comment by oliverdutton** [comment]
Implementing the XLA op on the backend directly.  This meant I didn't need the format changing to ONNX which is what pushed me to look at no XLA scatter.

**Pull request closed without merging**